### DISCUSSION TITLE
What would be the best way to have a (popup/window) on button click ?

### DISCUSSION THREAD

**Discussion by tom-avilius** [Aylur/ags#536] [Q&A]
Suppose I have a button showing network status. When I click on it I want a window where I could connect to a different network sort of thing. I want maximum control in placement of the window/widget. Wayland would not allow me to place the window to an (x, y) co-ordinate I think.

**Comment by Jas-SinghFSU** [comment]
Not sure what AGS can do since this is likely a GTK limitation? You can look at my implementation here that uses margins to spoof positioning: https://github.com/Jas-SinghFSU/HyprPanel/blob/master/modules/menus/shared/dropdown/locationHandler/index.ts

It's a function that takes the name of a window and [x, y] coordinates of the click, then creates margins for that dropdown to place the window at that location.

**Reply by zoenetic** [reply]
Do you know if there's a way to get x,y coordinates for an existing widget?
